src/app/modules/test-navigator/test-navigator.component.spec.ts

```diff
@@ -167,6 +168,49 @@ describe('TestNavigatorComponent', () => {
     expect(renameButton.nativeElement['title']).toEqual('cannot rename "test.tcl": unsaved changes');
   });
 
+  it('disables rename if the file of the selected node has unsaved changes',
+      fakeAsync(inject([MessagingService], async (messageBus: MessagingService) => {
+        await component.updateModel();
+        component.model.expanded = true;
+        component.model.children[1].dirty = false;
+        fixture.detectChanges();
+        const renameButton = fixture.debugElement.query(By.css('#rename'));
+        const testNode = fixture.debugElement.query(
+          By.css('app-tree-viewer > div > div:nth-child(2) > div:nth-child(2) .tree-view-item-key'));
+
+        // when
+        testNode.triggerEventHandler('dblclick', new MouseEvent('dblclick'));
+        messageBus.publish(EDITOR_DIRTY_CHANGED, { path: component.model.children[1].id, dirty: true });
+        tick();
+        fixture.detectChanges();
+
+        // then
+        expect(renameButton.nativeElement.disabled).toBeTruthy();
+        expect(renameButton.nativeElement['title']).toEqual('cannot rename "test.tcl": unsaved changes');
+      })));
+
+  it('reactivates rename if file of the selected node had unsaved changes, but editor was closed',
```

Comment:
That's the user discarding his changes – another case that should also reactivate rename is when the opened and changed file is save. Does that warrant a separate test case?